src/App/App.js

```diff
@@ -42,6 +43,12 @@ class App extends Component {
 		getAllMovies()
 			.then(data => this.setState({movies: data.movies}))
 			.catch(error => this.setState({error}));
+
+		if (this.state.savedState && this.state.userID) {
+			this.props.history.push(`/users/${this.state.userID}`);
+		} else if (!this.state.userID) {
+			this.props.history.push('/');
+		}
```

Comment:
added conditional in componentDidMount method to check what the savedState is, and render the page to Guest or User Home off of that